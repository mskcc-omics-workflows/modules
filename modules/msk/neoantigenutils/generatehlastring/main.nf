process NEOANTIGENUTILS_GENERATEHLASTRING {
    tag "$meta.id"
    label 'process_single'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigen-utils-base:1.3.0':
        'docker.io/mskcc/neoantigen-utils-base:1.3.0' }"

    input:
    tuple val(meta),  path(inputHLA)

    output:
    tuple val(meta), stdout,   emit: hlastring
    path "versions.yml",       emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    generateHLAString.sh -f ${inputHLA}


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        generateHLAstring: \$(echo \$(generateHLAString.sh -v))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
        echo "HLA-test:01,HLA-test2:02"
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            generateHLAstring: \$(echo \$(generateHLAString.sh -v))
        END_VERSIONS
    """
}
