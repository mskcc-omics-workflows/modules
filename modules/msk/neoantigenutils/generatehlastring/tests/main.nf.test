nextflow_process {

    name "Test Process NEOANTIGENUTILS_GENERATEHLASTRING"
    script "../main.nf"
    process "NEOANTIGENUTILS_GENERATEHLASTRING"

    tag "modules"
    tag "modules_nfcore"
    tag "neoantigenutils"
    tag "neoantigenutils/generatehlastring"

    test("neoantigenutils_generatehlastring - hla - string") {

        when {

            process {
                """

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['winners_hla_txt'], checkIfExists: true)
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }


    test("neoantigenutils_generatehlastring - hla - string - stub") {

        options "-stub"

        when {

            process {
                """

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('winners_hla_txt')
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
