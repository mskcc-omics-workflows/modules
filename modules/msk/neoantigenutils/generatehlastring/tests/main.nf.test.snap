{
    "neoantigenutils_generatehlastring - hla - string - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "HLA-test:01,HLA-test2:02\n"
                    ]
                ],
                "1": [
                    "versions.yml:md5,85e2db7a93782eb39ae80340a9728de2"
                ],
                "hlastring": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "HLA-test:01,HLA-test2:02\n"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,85e2db7a93782eb39ae80340a9728de2"
                ]
            }
        ],
        "timestamp": "2024-04-30T09:36:22.811416"
    },
    "neoantigenutils_generatehlastring - hla - string": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "HLA-A24:02,HLA-A24:02,HLA-B39:01,HLA-B39:01,HLA-C07:01,HLA-C06:02\n"
                    ]
                ],
                "1": [
                    "versions.yml:md5,85e2db7a93782eb39ae80340a9728de2"
                ],
                "hlastring": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "HLA-A24:02,HLA-A24:02,HLA-B39:01,HLA-B39:01,HLA-C07:01,HLA-C06:02\n"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,85e2db7a93782eb39ae80340a9728de2"
                ]
            }
        ],
        "timestamp": "2024-04-30T09:36:16.803412"
    }
}